network/dell/dnos10_config.py

```diff
@@ -0,0 +1,260 @@
+  retured: when not check_mode
+  type: list
+  sample: ['...', '...']
+"""
+from ansible.module_utils.netcfg import NetworkConfig, dumps, ConfigLine
+from ansible.module_utils.dnos10 import NetworkModule, dnos10_argument_spec
+from ansible.module_utils.dnos10 import get_config, get_sublevel_config 
+
+def get_candidate(module):
+    candidate = NetworkConfig(indent=1)
+    if module.params['src']:
+        candidate.load(module.params['src'])
+    elif module.params['lines']:
+        parents = module.params['parents'] or list()
+        candidate.add(module.params['lines'], parents=parents)
+    return candidate
+
+
+def main():
+
+    argument_spec = dict(
+        lines=dict(aliases=['commands'], type='list'),
+        parents=dict(type='list'),
+
+        src=dict(type='path'),
+
+        before=dict(type='list'),
+        after=dict(type='list'),
+
+        match=dict(default='line',
+                   choices=['line', 'strict', 'exact', 'none']),
+        replace=dict(default='line', choices=['line', 'block']),
+
+        update_config=dict(type='bool', default=True),
+        backup_config=dict(type='bool', default=False)
+    )
+    argument_spec.update(dnos10_argument_spec)
+
+    mutually_exclusive = [('lines', 'src')]
+
+    module = NetworkModule(argument_spec=argument_spec,
+                           connect_on_load=False,
+                           mutually_exclusive=mutually_exclusive,
+                           supports_check_mode=True)
+
+    module.check_mode = not module.params['update_config']
+
+    parents = module.params['parents'] or list()
+
+    match = module.params['match']
+    replace = module.params['replace']
+    before = module.params['before']
+    result = dict(changed=False, saved=False)
+
+    candidate = get_candidate(module)
+
+    if module.params['match'] != 'none':
+        config = get_config(module)
+        if parents:
+            contents = get_sublevel_config(config, module)
+            config = NetworkConfig(contents=contents, indent=1)
+        configobjs = candidate.difference(config, match=match, replace=replace)
+
+    else:
+        configobjs = candidate.items
+
+    if module.params['backup_config']:
+        result['__backup__'] = module.cli('show running-config')[0]
+
+    commands = list()
+    if configobjs:
+        commands = dumps(configobjs, 'commands')
+        commands = commands.split('\n')
+
+        if module.params['before']:
+            commands[:0] = module.params['before']
+
+        if module.params['after']:
+            commands.extend(module.params['after'])
+
+        if not module.check_mode:
+            response = module.config.load_config(commands)
+            result['responses'] = response
+
+            if module.params['save_config']:
+                module.config.save_config()
+                result['saved'] = True
+
+        result['changed'] = True
+
+    result['updates'] = commands
+    result['connected'] = module.connected
```

Comment:
not needed.  You can leave here if you want to but it needs to be documented in the RETURNS doc string